x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,404 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CloudSetup } from '@kbn/cloud-plugin/server';
+import { DeprecationsDetails } from '@kbn/core-deprecations-common';
+import { GetDeprecationsContext, RegisterDeprecationsConfig } from '@kbn/core-deprecations-server';
+
+import { hasFleetServers } from '@kbn/fleet-plugin/server/services/fleet_server';
+import { isAgentlessEnabled } from '@kbn/fleet-plugin/server/services/utils/agentless';
+import { i18n } from '@kbn/i18n';
+import { Connector, fetchConnectors } from '@kbn/search-connectors';
+
+import { ConfigType } from '..';
+
+export const getRegisteredDeprecations = (
+  config: ConfigType,
+  cloud: CloudSetup,
+  docsUrl: string
+): RegisterDeprecationsConfig => {
+  return {
+    getDeprecations: async (ctx: GetDeprecationsContext) => {
```

Comment:
You're not handling any exceptions in this function, will the upgrade assistant code call this `getDeprecations` function and handle rejected promises if any of the functions call in this function throw errors?

Reply:
I'm not handling them because I don't have a way to specify special handling. Instead, I'm letting any error raise. In my testing, this leads to an error banner, and the line-item gets a label like, "_not_ resolved" in red. Then you can click back into the row and try clicking the quick-resolve button again, or can try to address the issue.

Reply:
Cool. Then yeah, I think this is probably fine to let any errors bubble up.